pkg/daemon/ceph/mds/filesystem.go

```diff
@@ -71,15 +69,24 @@ func (f *Filesystem) CreateFilesystem(context *clusterd.Context, clusterName str
 		return fmt.Errorf("at least one data pool must be specified")
 	}
 
-	err = ceph.CreatePoolWithProfile(context, clusterName, *f.metadataPool, appName)
+	fslist, err := client.ListFilesystems(context, clusterName)
+	if err != nil {
+		return fmt.Errorf("Unable to list existing file systems. %+v", err)
+	}
+	if len(fslist) > 0 && !(client.IsMultiFSEnabled()) {
```

Comment:
maybe it's a style preference, but you don't need parens around `client.IsMultiFSEnabled()`

Reply:
Done.